core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -66,6 +78,17 @@
 
     private AppInsightsConfigurationBuilder builder = new JaxbAppInsightsConfigurationBuilder();
 
+    private static final Set<String> defaultPerformaceModuleClassNames = new HashSet<>();
+
+    static {
```

Comment:
Are you sure that this works. I think the static block executes before even the constructor of the program is called and it is when the constructor is called the memory is allocated to the variables. I might be wrong, please correct me if so.

Reply:
The static initializer (aka <clinit>) is executed whenever the class name is dereferenced. It runs exactly once.

What's the concern here?